src/app/plugins/stock.plugin.ts

```diff
@@ -0,0 +1,99 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, IHttpResponse } from '../../common/types';
+import Environment from '../../environment';
+
+export class StockPlugin extends Plugin {
+  public name: string = 'Stock Plugin';
+  public description: string = 'Get stock quotes';
+  public usage: string = 'stock <ticker>; stock AAPL';
+  public permission: ChannelType = ChannelType.Public;
+
+  private _API_URL: string = 'https://www.alphavantage.co/query?';
+  private _ADVANCED_QUOTE_LINK: string = 'https://finance.yahoo.com/quote/';
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public validate(message: IMessage, args: string[]) {
+    // in order to error handle
+    return args.length > 0;
```

Comment:
fyi: you can do `!args.length` since 0 is a falsey value in js

Reply:
less meaningful